neuralop/layers/embeddings.py

```diff
@@ -137,10 +153,64 @@ def forward(self, data, batched=True):
         out =  torch.cat((data, *grids),
                          dim=1)
         return out
+
+class NeRFEmbedding(Embedding):
+    """NeRFEmbedding implements the positional encoding from [1]_. 
+
+    Each input channel value p is embedded via a function g 
+        with 2L channels such that g(p) is a 2L-dim vector:
+            g(p)_k = sin(2^(k) * Pi * p) if i is odd
+                   = cos(2^(k) * Pi * p) if i is even
+    References
+    -----------
+    
+    .. _[1]: 
+    
+    Mildenhall, B. et al (2020). 
+        "NeRF: Representing Scenes as Neural Radiance Fields for View Synthesis".
+        ArXiv, https://arxiv.org/pdf/2003.08934. 
+    """
+    def __init__(self, num_frequencies):
+        super().__init__()
+        self.num_frequencies = num_frequencies
     
-class SinusoidalEmbedding2D(nn.Module):
+    def out_channels(self, channels):
+        return 2 * self.num_frequencies * channels
+
+    def forward(self, x):
+        """
+        x: torch.Tensor, shape (n_in, out_channels)
+        
+        """
+        assert x.ndim in [2,3], f"Error: expected inputs of shape (batch, n_in, channels)\
+            or (n_in, channels), got inputs with ndim=={x.ndim}, shape={x.shape}"
+        if x.ndim == 2:
+            batched = False
```

Comment:
I wonder if we can add some sort of warning here? Just to notify the user in case there was some mistake in their input where there was mistakenly one less dim. Not sure if this is necessary, just a thought.

Reply:
Not sure about this. Torch automatically adds a batch dim via `collate_fn` in most dataloaders, so anything not in this form was probably done intentionally by a user. @JeanKossaifi what do you think?